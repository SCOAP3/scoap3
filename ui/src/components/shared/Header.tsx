/* eslint-disable react/jsx-key */
import React, { useState } from "react";
import Image from "next/image";
import Link from "next/link";
import { Button } from "antd";
import { MenuOutlined } from "@ant-design/icons";

import { MenuItem } from "@/types";

interface MenuProps {
  items: MenuItem[];
  mobile?: boolean;
  collapsed?: boolean;
}

const labels = [
  <Link href="/">Home</Link>,
  <Link href="https://scoap3.org/" target="_blank" rel="noopener noreferrer">
    ScoapÂ³ project
  </Link>,
  <Link href="/partners">Partners</Link>,
  <a
    href="https://scoap3.org/scoap3-repository"
    target="_blank"
    rel="noopener noreferrer"
  >
    About
  </a>,
  <a
    href="https://scoap3.org/scoap3-repository-help"
    target="_blank"
    rel="noopener noreferrer"
  >
    Help
  </a>,
  <a
    href="https://github.com/SCOAP3/scoap3-next/wiki"
    target="_blank"
    rel="noopener noreferrer"
  >
    Documentation
  </a>,
  <Link href="/login">Login</Link>,
];

const Menu: React.FC<MenuProps> = ({
  items,
  mobile = false,
  collapsed = true,
}) => (
  <ul
    className={`menu-${mobile ? "mobile" : "desktop"} ant-menu ${
      collapsed ? "collapsed" : "visible"
    }`}
  >
    {items.map((item: MenuItem) => (
      <li className="ant-menu-item" key={item.key}>
        {item.label}
      </li>
    ))}
  </ul>
);

const Header: React.FC = () => {
  const [collapsed, setCollapsed] = useState<boolean>(true);

  const toggleCollapsed = () => {
    setCollapsed(!collapsed);
  };

  const headerItems: MenuItem[] = labels.map((label, index) => ({
    key: (index + 1).toString(),
    label,
  }));

  return (
    <div className="header">
      <div className="container flex items-center ">
        <Link className="logo" href="/">
          <Image
            src="/images/logo.png"
            width={140}
            height={40}
            alt="Logo of Scope3"
          />
        </Link>
        <Menu items={headerItems} />
        <Button onClick={toggleCollapsed}>{<MenuOutlined />}</Button>
      </div>
      <Menu mobile items={headerItems} collapsed={collapsed} />
    </div>
  );
};

export default Header;
