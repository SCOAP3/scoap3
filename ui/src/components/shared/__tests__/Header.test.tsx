import React from "react";
import { render } from "@testing-library/react";

import Header from "../Header";

describe("Header", () => {
  it("renders header correctly", () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
