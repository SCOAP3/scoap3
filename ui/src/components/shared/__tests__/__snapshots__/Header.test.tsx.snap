// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header correctly 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="container flex items-center "
    >
      <a
        class="logo"
        href="/"
      >
        <img
          alt="Logo of Scope3"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Flogo.png&w=384&q=75"
          srcset="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=384&q=75 2x"
          style="color: transparent;"
          width="140"
        />
      </a>
      <ul
        class="menu-desktop ant-menu collapsed"
      >
        <li
          class="ant-menu-item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="https://scoap3.org/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Scoap³ project
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="/partners"
          >
            Partners
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="https://scoap3.org/scoap3-repository"
            rel="noopener noreferrer"
            target="_blank"
          >
            About
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="https://scoap3.org/scoap3-repository-help"
            rel="noopener noreferrer"
            target="_blank"
          >
            Help
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="https://github.com/SCOAP3/scoap3-next/wiki"
            rel="noopener noreferrer"
            target="_blank"
          >
            Documentation
          </a>
        </li>
        <li
          class="ant-menu-item"
        >
          <a
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
      <button
        class="ant-btn css-dev-only-do-not-override-2q8sxy ant-btn-default"
        type="button"
      >
        <span
          aria-label="menu"
          class="anticon anticon-menu"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="menu"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </button>
    </div>
    <ul
      class="menu-mobile ant-menu collapsed"
    >
      <li
        class="ant-menu-item"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="ant-menu-item"
      >
        <a
          href="https://scoap3.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Scoap³ project
        </a>
      </li>
      <li
        class="ant-menu-item"
      >
        <a
          href="/partners"
        >
          Partners
        </a>
      </li>
      <li
        class="ant-menu-item"
      >
        <a
          href="https://scoap3.org/scoap3-repository"
          rel="noopener noreferrer"
          target="_blank"
        >
          About
        </a>
      </li>
      <li
        class="ant-menu-item"
      >
        <a
          href="https://scoap3.org/scoap3-repository-help"
          rel="noopener noreferrer"
          target="_blank"
        >
          Help
        </a>
      </li>
      <li
        class="ant-menu-item"
      >
        <a
          href="https://github.com/SCOAP3/scoap3-next/wiki"
          rel="noopener noreferrer"
          target="_blank"
        >
          Documentation
        </a>
      </li>
      <li
        class="ant-menu-item"
      >
        <a
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </div>
</div>
`;
